synapse/storage/databases/main/stats.py

```diff
@@ -773,7 +778,7 @@ def get_users_media_usage_paginate_txn(
 
             args += [limit, start]
             txn.execute(sql, args)
-            users = self.db_pool.cursor_to_dict(txn)
+            users = cast(List[Tuple[str, Optional[str], int, int]], txn.fetchall())
```

Comment:
Hoooo boy. Alright, the selection is roughly

```
SELECT
    lmr.user_id,
    displayname,
    COUNT(lmr.user_id) as media_count,
    SUM(media_length) as media_length
FROM
    local_media_repository as lmr
    LEFT JOIN profiles AS p ON lmr.user_id = p.full_user_id
    {where_clause}
    GROUP BY lmr.user_id, displayname
```

and the relevant bits (other columns omitted) of the table schemas are

```
matrix=> \d local_media_repository
              Table "matrix.local_media_repository"
        Column        │  Type   │ Collation │ Nullable │ Default 
══════════════════════╪═════════╪═══════════╪══════════╪═════════
 user_id              │ text    │           │          │ 
 media_length         │ integer │           │          │  

matrix=> \d profiles
               Table "matrix.profiles"
    Column    │ Type │ Collation │ Nullable │ Default 
══════════════╪══════╪═══════════╪══════════╪═════════
 displayname  │ text │           │          │ 
 full_user_id │ text │           │          │ 
```

Reply:
#16545 captures the `local_media_repository` bits.

Reply:
- `lmr.user_id` is nullable and on the left side of the join, so I think the first result column may be null. (But I hope to $DEITY that it never is in practice).
- displayname is nullable (as the type hint says).
- the COUNT is never null
- but the SUM may be, if it is a sum of zero rows. From https://www.postgresql.org/docs/current/functions-aggregate.html
  > It should be noted that except for count, these functions return a null value when no rows are selected. In particular, sum of no rows returns null, not zero as one might expect, and array_agg returns null rather than an empty array when there are no input rows. The coalesce function can be used to substitute zero or an empty array for null when necessary.

Reply:
> #16545 captures the `local_media_repository` bits.

Gah, I could did try to search for this... but evidently not hard enough.

Reply:
> but the SUM may be, if it is a sum of zero rows.

Ahh, but there never will be, it's a sum of the rows in a group.

SGTM!